nixos/modules/services/cluster/kubernetes/addons/dns.nix

```
@@ -60,6 +60,45 @@ in {
         sha256 = "02r440xcdsgi137k5lmmvp0z5w5fmk8g9mysq5pnysq1wl8sj6mw";
       };
     };
+
+    corefile = mkOption {
+      description = ''
+        Custom coredns corefile configuration.
+
+        See: <link xlink:href="https://coredns.io/manual/toc/#configuration"/>.
+      '';
+      type = types.str;
+      default = "
+        .:${toString ports.dns} {
+          errors
+          health :${toString ports.health}
+          kubernetes ${cfg.clusterDomain} in-addr.arpa ip6.arpa {
+            pods insecure
+            fallthrough in-addr.arpa ip6.arpa
+          }
+          prometheus :${toString ports.metrics}
+          forward . /etc/resolv.conf
+          cache 30
+          loop
+          reload
+          loadbalance
+        }";
+      defaultText = "
+        .:${toString ports.dns} {
+          errors
+          health :${toString ports.health}
+          kubernetes \${cfg.clusterDomain} in-addr.arpa ip6.arpa {
```

I think it might be better to write:

```suggestion
          kubernetes \${config.services.kubernetes.addons.dns.clusterDomain} in-addr.arpa ip6.arpa {
```

so that the example shown in the manual can be literally pasted into your own config (currently `cfg` would not be in scope for that).

(For that it'd be handy if the `ports` were also a config options, so we could do the same there, but unfortunately they aren't, so we can't and changing that is out of the scope of this PR.)

---

Changed.